Dnn.AdminExperience/ClientSide/Security.Web/src/components/apiTokenSettings/index.jsx

```diff
@@ -0,0 +1,199 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import { security as SecurityActions } from "../../actions";
+import {
+    Dropdown,
+    InputGroup,
+    SingleLineInputWithError,
+    Switch,
+    Label,
+    Button,
+    Tooltip,
+} from "@dnnsoftware/dnn-react-common";
+import "./style.less";
+import util from "../../utils";
+import resx from "../../resources";
+import styles from "./style.less";
+
+class ApiTokenSettingsPanelBody extends Component {
+    constructor() {
+        super();
+        this.state = {
+            apiTokenSettings: undefined,
+        };
+    }
+
+    UNSAFE_componentWillMount() {
```

Comment:
I'm not a React expert, but I'm surprised to see `UNSAFE` in a new component. Is this because it's inheriting from the shared components which require this?

Reply:
This is copy-paste from another component. But we are several versions behind in React and this is part of legacy support if I'm correct.